File: 2021/c/1/part1.c
#include <stdio.h>

int main(int argc, char** argv)
{
        FILE *f;
        int depth;
        int ldepth = -1;
        int n = -1;

        f = fopen("input.txt", "r");
        if (!f) {
                return 1;
        }

        while (1 == fscanf(f, "%d", &depth)) {
                if (depth > ldepth) {
                        ++n;
                }
                ldepth = depth;
        }

        printf("%d\n", n);
        fclose(f);
        return 0;
}

File: 2021/rs/1/part1.rs
use std::fs;
use std::io;
use std::io::prelude::*;

fn main() -> io::Result<()> {
    let file = fs::File::open("input.txt")?;
    let br = io::BufReader::new(file);

    let mut cnt: i32 = -1;
    let mut last_depth: i32 = -1;
    for line in br.lines() {
        let depth: i32 = line
            .expect("Failed to read line.")
            .trim()
            .parse()
            .expect("Failed to parse line.");
        if depth > last_depth {
            cnt = cnt + 1;
        }
        last_depth = depth;
    }

    println!("{}", cnt);

    Ok(())
}
